#!/usr/bin/env python
PACKAGE = "state_estimation"
# The following line should not be here for Groovy and above
import roslib;roslib.load_manifest(PACKAGE)

# For Groovy and above, the following line should be
# from dynamic_reconfigure.parameter_generator_catkin import *
from dynamic_reconfigure.parameter_generator import *

gen = ParameterGenerator()

gen.add("tf_inverse",   bool_t,   0, "TF Inverse",  False)
gen.add("x", double_t, 0, "X position", 0, -1.0, 1.0)
gen.add("y", double_t, 0, "Y position", 0, -1.0, 1.0)
gen.add("z", double_t, 0, "Z position", 0, -1.0, 1.0)

rot_enum = gen.enum([ gen.const("Euler", int_t, 0, "Euler"),
                  gen.const("Quaternion", int_t, 1, "Quaternion") ],
                  "Rotation type (euler/quaternion)")

gen.add("rot_type", int_t, 0, "Rotation type (euler/quaternion)", 0, 0, 1, edit_method=rot_enum)

gen.add("roll", double_t, 0, "Roll", 0, -180.0, 180.0)
gen.add("pitch", double_t, 0, "Pitch", 0, -180.0, 180.0)
gen.add("yaw", double_t, 0, "Yaw", 0, -180.0, 180.0)

gen.add("qx", double_t, 0, "qX", 0, -1.0, 1.0)
gen.add("qy", double_t, 0, "qY", 0, -1.0, 1.0)
gen.add("qz", double_t, 0, "qZ", 0, -1.0, 1.0)
gen.add("qw", double_t, 0, "qW", 1, -1.0, 1.0)

exit(gen.generate(PACKAGE, "pose_generator", "PoseGen"))
